Makefile

```diff
@@ -10,8 +10,17 @@ SHIM ?= "${GOPATH}/src/github.com/clearcontainers/shim/cc-shim"
 # The time limit in seconds for each test
 TIMEOUT ?= 5
 
-functional:
-	ginkgo ./functional/ -- -runtime "${RUNTIME}" -proxy "${PROXY}" -shim "${SHIM}" -timeout ${TIMEOUT}
+TESTS_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+GINKGO_PATH = ${GOPATH}/bin/ginkgo
+
+ginkgo:
+ifeq ($(wildcard $(GINKGO_PATH)),)
```

Comment:
I agree that some of the make functions can be pretty opaque, but I'm used to this idiom personally. However, how about changing it to use `$(realpath )` since that make is _slightly_ clearer what is going on.

Alternatively, you could do something like the following which makes the logic clearer still:

```
# Return non-empty string if file exists
define file_exists
$(realpath $1)
endef

exists = $(call file_exists,$(GINKGO_PATH))

ginkgo:
ifeq ($(exists),)
  go get github.com/onsi/ginkgo/ginkgo
endif

```